### ISSUE TITLE
SIMPLE: option to display activity using a simpler layout

### ISSUE THREAD

**Issue by paul90** [fedwiki/wiki-plugin-activity#30]
I am splitting the layout aspect from #28, raised by @opn, so it receives the separate attention it deserves.

---

...the rendering of the results is unnecessarily complex / ugly for a minimal presentation:

```
Page Activity
  since Wed Aug 10 2016
  excluding neighborhood
```

For a minimal presentation we just need the title and a list of links. Possibly/optionally including the data of the change. With the following markup:

```
NEIGHBORHOOD no
NARRATIVE
LAST 3
```

Rendering something equivalent (in markdown) to:

```
# Recent changes to this site
*  [[Some Page]] - Aug 10 2016
*  [[Another Page]] - Aug 9 2016
*  [[Last Page]] - Jan 3 2016
```

**Comment by paul90** [comment]
There has already been some discussion in #28, but it looked as if the discussion there was starting to get entangled.

## Activity Title
The activity title is without doubt too wordy. There is a thought to remove it, and leaving an explanation of the activity content to the page author.

Any activity title is best contained within the plug-in output. And suggest a new `TITLE` verb to allow for both a user written title, or turning it off. The user written title could allow a placeholder to include the `SINCE` date, if used.

It should not impossible to devise better generated titles.

## `SIMPLE` Layout

The suggestion is written starting from suggesting a new verb `LAST`. It is probably worth splitting this up into it's parts, as some of these might be useful in other contexts.

Using the example : 
```
NEIGHBORHOOD no
LAST 3
```
``` markdown
# Recent changes to this site
*  [[Some Page]] - Aug 10 2016
*  [[Another Page]] - Aug 9 2016
*  [[Last Page]] - Jan 3 2016
```
The output starts with a custom `TITLE` that is appropriate to `LAST`. A sensible default set of words could be used by default, and could be overridden by the user if they want.

The output does not contain any time period based subheadings. There are probably some instances with `SINCE` where disabling these sub-headings, or some of them, might also be of interest. *The ones that spring to mind is `SINCE` 1 day, and 1 week, where the next sub-head can appear.*

Turning off FLAGS, and optionally showing dates, this only makes sense if the active is only for a single site. But, is there a need to allow customization of date format?

**Comment by WardCunningham** [comment]
I like the pattern where simple markup is echoed as part of rendered output. The activity plugin may have already missed this opportunity. Adding a command to make it simpler is a daring move. Would we expect this command to be echoed too?

The markup as heading pattern works better when it is joined to the output by having both surrounded by a gray background.

**Comment by opn** [comment]
Title suggestion is good.

> And suggest a new TITLE verb to allow for both a user written title, or turning it off. The user written title could allow a placeholder to include the SINCE date, if used.

`TITLE none`

should turn off all titles - including the inner date specific subtitles? This may be enough as we can always add our own markdown or HTML title in the line above? But:

`TITLE Recent changes to this site`

would work for a custom title if that is not too much work?

> But, is there a need to allow customization of date format?

I don't think that is needed. Indeed the option for dates with the LAST option is not vital. If the date option is added we would need to be able to turn it off as the entries will often be too long to fit on a line which can be ugly - so often it may be useful to display a simple minimal list of last changes without dates - so that it looks minimal and like a minimal list of contents, with each title on a single line. Naturally if a page has a long title it won;t fit on a single line - but since many authors craft their titles to fit nicely we should have a style that preserves this layout elegantly?

> Turning off FLAGS, and optionally showing dates, this only makes sense if the active is only for a single site. 

Why does it make sense only for a single site? If I have a ROSTER that lists three sites then:

```
ROSTER all
LAST 3
```
would use the sitemap to sort entries by date and show the three most recent entries - which may all come from one site or a mix - but the logic makes sense we are interested only in the last 3 chronologically sorted changes.

**Comment by WardCunningham** [comment]
I have a bad feeling about where this is going. Perhaps it is time to stop tweeking one representation and consider new representations.

Let's use all of the information in the Journal and paint an accurate picture of what's happened, not just what happened recently. I'm a fan of Dots that Merge.

![image](https://cloud.githubusercontent.com/assets/12127/17786811/67b513a6-653b-11e6-94ec-bf2b9bf94d8d.png)


http://ward.bay.wiki.org/dots-that-merge.html
